Source/ARTWebSocketTransport.m

```diff
@@ -114,7 +114,14 @@ - (void)connectForcingNewToken:(BOOL)forceNewToken {
 
                 if (error) {
                     [selfStrong.logger error:@"R:%p WS:%p ARTWebSocketTransport: token auth failed with %@", _delegate, self, error.description];
-                    [selfStrong.delegate realtimeTransportFailed:selfStrong withError:[[ARTRealtimeTransportError alloc] initWithError:error type:ARTRealtimeTransportErrorTypeAuth url:self.websocketURL]];
+                    if (error.code == 40102 /*incompatible credentials*/) {
+                        // RSA15c
+                        [selfStrong.delegate realtimeTransportFailed:selfStrong withError:[[ARTRealtimeTransportError alloc] initWithError:error type:ARTRealtimeTransportErrorTypeAuth url:self.websocketURL]];
+                    }
+                    else {
+                        // RSA4b
+                        [selfStrong.delegate realtimeTransportDisconnected:selfStrong withError:[[ARTRealtimeTransportError alloc] initWithError:error type:ARTRealtimeTransportErrorTypeAuth url:self.websocketURL]];
```

Comment:
If you're referring to https://github.com/ably/ably-ios/pull/518#discussion-diff-85763133R216 then that's a different spec